### PR TITLE
Use forceUpdate rather than setState(state)

### PR DESCRIPTION
**Pull request by jillesme**
Currently you use this.setState(state) to re-render, there's a method for that (`this.forceUpdate()`);

Hope it helps, cheers!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ var Notify = React.createClass({
 	addNotify: function(title, msg, time, theme) {
 		var key = this.key++;
 		this.state[key] = { title: title, msg: msg, time: time, theme: theme };
-		this.setState(this.state);
+		this.forceUpdate();
```
Comment (philipjc) [inline index.js:20]: You may want to rethink this, you only want to force update if you make changes and don't want to set the state. But here the state is being changed, so set state is correct. Force update will ignore shouldComponentUpdate.
https://facebook.github.io/react/docs/component-api.html#forceupdate

Comment (jillesme) [inline index.js:20]: I believe you are wrong. He was using this.setState(this.state); to trigger a re-render. That is literally what `this.forceUpdate()` does. 

From the docs:

`you can tell React that it needs to re-run render() by calling forceUpdate().`

Comment (ryanclark) [inline index.js:20]: Realistically you shouldn't directly be modifying `this.state`. If you do, there isn't really much difference in calling `this.setState` or `this.forceUpdate`. 

Alternatively, consider passing a function through to `this.setState` -

``` js
this.setState(function (prevState, props) {
   prevState[key] = {}; // whatever here
   return prevState;
});
```

**Comment by jillesme** [comment]
I misunderstood why you were using this.setState(this.state); I thought it was just to trigger a re-render. Which isn't the case,

**Pull request closed without merging**